front/src/components/Register.js

```diff
@@ -19,153 +19,161 @@ const style = {
 
 class Register extends Component {
 
-  state = {
-    username: "",
-    password: "",
-    username_error_text: null,
-    password_error_text: null,
-    disabled: true,
-  };
-
-  onSubmit = e => {
-    e.preventDefault();
-    this.props.register(this.state.username, this.state.password);
-  };
-
-  validateUsername = username => {
-    const re = /^/;
-    return re.test(username);
-  };
-
-  isDisabled() {
-    let username_is_valid = false;
-    let password_is_valid = false;
-
-    if (this.state.username === '') {
-        this.setState({
-            username_error_text: null,
-        });
-    } else if (this.validateUsername(this.state.username)) {
-        username_is_valid = true;
-        this.setState({
-            username_error_text: null,
-        });
-
-    } else {
-        this.setState({
-            username_error_text: 'Sorry, this is not a valid username',
-        });
-    }
+    state = {
+        username: "",
+        password: "",
+        username_error_text: null,
+        password_error_text: null,
+        disabled: true,
+    };
+
+    onSubmit = e => {
+        e.preventDefault();
+        this.props.register(this.state.username, this.state.password);
+    };
+
+    validateUsername = username => {
+        const re = /^(?![_ -])(?:(?![_ -]{2})[\w -]){5,16}(?<![_ -])$/;
```

Comment:
Lets make endpoint for validation `username` on backend. After all we will have validation in one place)

Reply:
Yes, it will be added in the future. But this validation is also needed in order not to make a request to immediately inform the client about the incorrectness of the entered name or password.